### PR TITLE
[feat]: [PIE-5352]: OPA: Restrict execution to main Branch

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Customer use cases:
InMobi - customer would like to block all deployment/builds that are triggered from any branches other than the main or master branch.

Solution:
To add gitInfo to expanded pipeline Json for Policy Evaluation

Changes:
Added a boolean which keeps tracks whether the method "fetchExpandedPipelineJSONFromYaml" is triggered during execute flow only, and in "fetchExpandedPipelineJSONFromYaml" i've added gitConfigDetails in the expandedYaml for execute flow "only". For that a new class "PipelineGovernanceGitConfig" is created to club required fields.

Testing: I've checked verified my API flow in local. Further Testing will be done in PRE-QA(as PR-ENV didn't had  PolicyManager)

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java [hunk 1]
```diff
@@ -489,4 +507,30 @@ private boolean isAlreadyImported(String accountIdentifier, String repoURL, Stri
...
   }
+
+  ExpansionResponseBatch getGitDetailsAsExecutionResponse(ScmGitMetaData scmGitMetaData) {
+    PipelineGovernanceGitConfig pipelineGovernanceGitConfig =
+        getPipelineGovernanceGitConfigInfoFromgitEntityInfo(scmGitMetaData);
+
+    String gitDetailsJson = JsonUtils.asJson(pipelineGovernanceGitConfig);
+    ExpansionResponseProto gitConfig = ExpansionResponseProto.newBuilder()
+                                           .setFqn(YAMLFieldNameConstants.PIPELINE)
+                                           .setKey("gitConfig")
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java:518]: Add string constant for gitConfig

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 2]
```diff
@@ -282,7 +282,8 @@ && isEmpty(triggerDetails.getNgTriggerConfigV2().getInputSetRefs())) {
         pipelineEnforcementService.validateExecutionEnforcementsBasedOnStage(pipelineEntity);
 
         String expandedJson = pmsPipelineServiceHelper.fetchExpandedPipelineJSONFromYaml(pipelineEntity.getAccountId(),
-            pipelineEntity.getOrgIdentifier(), pipelineEntity.getProjectIdentifier(), pipelineYamlWithTemplateRef);
+            pipelineEntity.getOrgIdentifier(), pipelineEntity.getProjectIdentifier(), pipelineYamlWithTemplateRef,
+            false);
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:286]: It should be true here. Trigger is also an execution flow

**Review by namanvermah — CHANGES_REQUESTED** [review]

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: 8 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 298 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJBT6EyY7CVTGghA-:-NBvJBT7-zX14WMWIKxe:b-qwv5na) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L298)):*
> ```Java
>         jsonExpander.fetchExpansionResponses(expansionRequests, expansionRequestMetadata);
> 
>     ScmGitMetaData scmGitMetaData = GitAwareContextHelper.getScmGitMetaData();
> ```

this can be moved to if condition as not needed if the condition is not true



_Code quote:_
```Java
scmGitMetaData = GitAwareContextHelper.getScmGitMetaData()
```
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 513 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJ_Uc8Ay5gBGr95vL:-NBvJ_UdDIrRhBuL6kA3:bdymsgo) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L513)):*
> ```Java
>   ExpansionResponseBatch getGitDetailsAsExecutionResponse(ScmGitMetaData scmGitMetaData) {
>     PipelineGovernanceGitConfig pipelineGovernanceGitConfig =
>         getPipelineGovernanceGitConfigInfoFromgitEntityInfo(scmGitMetaData);
> ```

Too big a method name, can be getPipelineGovernanceGitConfigInfo ?
rest details you can add as comment over the method what does it do



_Code quote:_
```Java
getPipelineGovernanceGitConfigInfoFromgitEntityInfo
```
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java` line 589 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJzd60LYf2iH5uFT7:-NBvJzd60LYf2iH5uFT8:b92h2kk) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java#L589)):*
> ```Java
> 
>     return pmsPipelineServiceHelper.fetchExpandedPipelineJSONFromYaml(
>         accountId, orgIdentifier, projectIdentifier, pipelineEntityOptional.get().getYaml(), false);
> ```

Above method is called from Resourec Layer
@namanvermah should we take flag value from API with default value as false?



_Code quote:_
```Java
 false
```

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: 8 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 298 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJBT6EyY7CVTGghA-:-NBvMfeU8F9bt8n-Avvs:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L298)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

this can be moved to if condition as not needed if the condition is not true
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 513 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJ_Uc8Ay5gBGr95vL:-NBvMgQw0dSQcCOyUT-a:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L513)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Too big a method name, can be getPipelineGovernanceGitConfigInfo ?
rest details you can add as comment over the method what does it do
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java` line 518 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBlQz03GgmP8TIU6NR6-r2-518:-NBvMhBWEiU6SoL8gUd9:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceHelper.java#L518)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

Add string constant for gitConfig
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java` line 589 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJzd60LYf2iH5uFT7:-NBvMkHY875XPRYdzc82:bq9bjsi) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java#L589)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Above method is called from Resourec Layer
@namanvermah should we take flag value from API with default value as false?
</blockquote></details>

i think this flag value is only for Internal purpose, has nothing to do with user, would be better, if we keep it in methods internally, rather than taking it to top layer
___
*[`pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java` line 286 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBr_huaVK6wnd_nJp7--r2-286:-NBvN7NK5UnJE-Vx_jqg:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java#L286)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

It should be true here. Trigger is also an execution flow
</blockquote></details>

Done.

**Review by sahilhindwani-harness — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 4 of 7 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: 3 file(s) to review. 2 change requests. 5. unresolved discussions. 1 of 2 approvals obtained.

**Review by namanvermah — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 4 of 7 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: 1 change requests. 3. unresolved discussions. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java` line 589 at r2](https://reviewable.io/reviews/harness/harness-core/37396#-NBvJzd60LYf2iH5uFT7:-NBvXx7P4ahTgE5QXfrc:b-nt9iuh) ([raw file](https://github.com/harness/harness-core/blob/2c05e990813aebf12f6cb8b02d6379dc8dddd824/pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java#L589)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

i think this flag value is only for Internal purpose, has nothing to do with user, would be better, if we keep it in methods internally, rather than taking it to top layer
</blockquote></details>

Agree with Vivek here, this flag is true for execution flows and false everytime else. So no need for a separate API flag.

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 3 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by namanvermah — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37396)* status: :shipit: complete! ✓

**Pull request merged**